pandas/tseries/period.py

```diff
@@ -211,79 +247,16 @@ def _generate_range(cls, start, end, periods, freq, fields):
         return subarr, freq
 
     @classmethod
-    def _from_arraylike(cls, data, freq, tz):
-        if not isinstance(data, (np.ndarray, PeriodIndex,
-                                 DatetimeIndex, Int64Index)):
-            if lib.isscalar(data) or isinstance(data, Period):
-                raise ValueError('PeriodIndex() must be called with a '
-                                 'collection of some kind, %s was passed'
-                                 % repr(data))
+    def _simple_new(cls, data, name=None, freq=None, **kwargs):
```

Comment:
no I don't think it would work. You have to tell pickle what to do. E.g. we want to pickle only *certain* attributes and the data, nothing more. I don't think it works correctly otherwise (it tries to do too much).

You can try removing them but (aside from back-compat) I don't think it would work, but I could be wrong.

In any event I don't see it as a big deal; you want custom actions and that's what the special methods are for.

otherwise its *too* presumptous.